### ISSUE TITLE
use YAML instead of $ prefix for dynamic variables?

### ISSUE THREAD

**Issue by andreynering** [go-task/task#46]
For consistency with `^` being deprecated in favor of `task: task-name` syntax.

Before:

```yml
# Taskvars.yml
GIT_COMMIT: $git log -n 1 --format=%h
```

After:

```yml
# Taskvars.yml
GIT_COMMIT:
  sh: git log -n 1 --format=%h
```

Feedback welcome

**Comment by smyrman** [comment]
I agree with this proposal.

It also optionally allows for later features, like allowing an "env" section to be passed in (not necessarily suggesting that we do that now).

**Comment by smyrman** [comment]
For consistency, the  same syntax _could_ (perhaps also later) be allowed in cmds (or even deps I guess, if you want to run some shell commands in parallel as dependencies...).

**Comment by zbindenren** [comment]
If the following Taskvars.yaml would save me a lot of boilerplate:

```yaml
VAR1: hello
VAR2: {{.VAR1}} world #<- VAR2 would be hello world
```

This is probably not easly possible.

Also: `Taskvars_{GOOS}.yaml` would be cool to overwrite dynamic variables.

**Comment by smyrman** [comment]
@zbindenren, I don't see that as relevant to this issue. You can create a new issue for it perhaps, but as you say, it's not really easily possible to do this in a deterministic way.

Issue #40 is more closely related to what you ask.

**Comment by smyrman** [comment]
I guess it's _somewhat_ related, in that if variables where changed to be a list of objects, then it would be possible.
```yaml
vars:
  - VAR1: hello
  - VAR2: "{{.VAR1}} world"
  - VAR3: hey VAR4: there. # Also possible, but here VAR4 may not use VAR3...
```

**Comment by andreynering** [comment]
`Taskfile_{{.GOOS}}.yml` should be very simple.

About interpolation with same-context variables, I don't how to implement that. Maybe a list as @smyrman suggested, but seems a bit counter-intuitive to the user.

**Comment by smyrman** [comment]
>  Maybe a list as @smyrman suggested, but seems a bit counter-intuitive to the user.

I agree to that. It's not a good solution.

**Comment by andreynering** [comment]
@zbindenren `Taskvars_{{GOOS}}.yml` should be working on master branch.

**Comment by zbindenren** [comment]
Great thx :-)
